samples/materials/gltf2Trans.mat

```diff
@@ -0,0 +1,73 @@
+material {
+    name : Lit,
+    requires : [
+        uv0
+    ],
+    shadingModel : lit,
+    blending : transparent,
+    transparency : default,
+    parameters : [
+        {
+            type : sampler2d,
+            name : baseColorMap
+        },
+        {
+            type : float4,
+            name : baseColorFactor
+        },
+        {
+            type : sampler2d,
+            name : metallicRoughnessMap
+        },
+        {
+            type : float,
+            name : metallicFactor
+        },
+        {
+            type : float,
+            name : roughnessFactor
+        },
+        {
+            type : sampler2d,
+            name : normalMap
+        },
+        {
+            type : float,
+            name : normalScale
+        },
+        {
+            type : sampler2d,
+            name : aoMap
+        },
+        {
+            type : float,
+            name : aoStrength
+        },
+        {
+            type : sampler2d,
+            name : emissiveMap
+        },
+        {
+            type : float3,
+            name : emissiveFactor
+        }
+    ],
+}
+
+fragment {
+    void material(inout MaterialInputs material) {
+        material.normal = texture(materialParams_normalMap, getUV0()).xyz * 2.0 - 1.0;
+        material.normal.y = -material.normal.y;
+        prepareMaterial(material);
+
+        material.baseColor = texture(materialParams_baseColorMap, getUV0());
+        material.baseColor.rgb *= materialParams.baseColorFactor.xyz;
+        material.baseColor.rgb *= material.baseColor.a;
```

Comment:
I wonder if you should premultiply *before* applying the baseColorFactor.

Reply:
Shouldn't the result be the same considering that multiplication is commutative?

Reply:
Yes. It was a stupid comment on my part :))